pyccel/ast/variable.py

```diff
@@ -883,16 +883,19 @@ def __init__(self, base, *indices):
         shape = base.shape
         rank  = base.rank
 
-        # Add empty slices to fully index the object
-        if len(indices) < rank:
-            indices = indices + tuple([Slice(None, None)]*(rank-len(indices)))
-
-        if any(not isinstance(a, (int, TypedAstNode, Slice)) for a in indices):
+        if any(not isinstance(a, (int, TypedAstNode, Slice, LiteralEllipsis)) for a in indices):
             errors.report("Index is not of valid type",
                     symbol = indices, severity = 'fatal')
 
-        self._indices = tuple(LiteralInteger(a) if isinstance(a, int) else a for a in indices)
-        super().__init__()
+        if len(indices) == 1 and isinstance(indices[0], LiteralEllipsis):
+            self._indices = tuple(LiteralInteger(a) if isinstance(a, int) else a for a in indices)
+            indices = [Slice(None,None)]*rank
```

Comment:
Because I don't want to change indices and lose the information, I want to keep it as LiteralEllipsis , at the same time I want to calculate the right `rank`, so if I change `self._indices` to  `[Slice(None,None)]*rank` then we will lose the information, and it will generate the wrong code when we have a template of different numpy array ranks

Reply:
> it will generate the wrong code when we have a template of different numpy array ranks

Presumably you are specifically talking about the Python code?

Reply:
yes, I had this issue in the Python printer

Reply:
In that case I agree, although it seems a shame to have so much code duplication for Python